client/src/pages/locations/GeoLocation.js

```diff
@@ -0,0 +1,122 @@
+import React from "react"
+import PropTypes from "prop-types"
+import { Button, Col, ControlLabel, FormGroup } from "react-bootstrap"
+import { Field } from "formik"
+import { Location } from "models"
+import REMOVE_ICON from "resources/delete.png"
+import * as FieldHelper from "components/FieldHelper"
+
+export const GEO_LOCATION_DISPLAY_TYPE = {
+  FORM_FIELD: "FORM_FIELD",
+  GENERIC: "GENERIC"
+}
+
+const GeoLocation = ({
+  lat,
+  lng,
+  setFieldTouched,
+  setFieldValue,
+  isSubmitting,
+  editable,
+  displayType
+}) => {
+  const setTouched = touched => {
+    setFieldTouched("lat", touched, false)
+    setFieldTouched("lng", touched, false)
+  }
+
+  if (!editable) {
+    const humanValue = (
+      <>
+        <span>{Location.parseCoordinate(lat) || "?"}</span>
+        <span>,&nbsp;</span>
+        <span>{Location.parseCoordinate(lng) || "?"}</span>
+      </>
+    )
+
+    if (displayType === GEO_LOCATION_DISPLAY_TYPE.GENERIC) return humanValue
+
+    if (displayType === GEO_LOCATION_DISPLAY_TYPE.FORM_FIELD) {
+      return (
+        <Field
+          name="location"
+          label="Latitude, Longitude"
+          component={FieldHelper.ReadonlyField}
+          humanValue={humanValue}
+        />
+      )
+    }
+
+    throw new Error(`Unknown displayType: '${displayType}' for GeoLocation`)
+  }
+
+  return (
+    <FormGroup style={{ marginBottom: 0 }}>
+      <Col sm={2} componentClass={ControlLabel} htmlFor="lat">
+        Latitude, Longitude
+      </Col>
+
+      <Col sm={7}>
+        <Col sm={3} style={{ marginRight: "8px" }}>
+          <Field
+            name="lat"
+            component={FieldHelper.InputFieldNoLabel}
+            onBlur={() => {
+              setTouched(true)
+              setFieldValue("lat", Location.parseCoordinate(lat))
```

Comment:
`setFieldValue` is called unconditionally, yet according to the propTypes it is not required, and there's no default prop value supplied.

Reply:
Added custom propType validation for `setFieldValue`.